catalog/dags/common/storage/media.py

```diff
@@ -304,17 +303,25 @@ def _format_raw_tag(self, tag: str) -> dict:
 
     def _validate_filetype(self, filetype: str | None, url: str) -> str | None:
         """
-        Extract filetype from the media URL if filetype is None.
+        Extract filetype from the media URL if filetype is None and check
+        that it corresponds to the media type of the class, while also unifies
+        filetypes that have variants such as jpg/jpeg and tiff/tif.
 
-        Unifies filetypes that have variants such as jpg/jpeg and tiff/tif.
         :param filetype: Optional filetype string.
+        :param url: The direct URL to the media.
         :return: filetype string or None
         """
-        if filetype is None:
-            filetype = extract_filetype(url, self.media_type)
-        if self.media_type != "image":
-            return filetype
-        return FILETYPE_EQUIVALENTS.get(filetype, filetype)
+        if filetype is not None:
+            filetype = filetype.lower()
+            return FILETYPE_EQUIVALENTS.get(filetype, filetype)
+
+        filetype, extracted_media_type = extract_filetype(url)
+        if extracted_media_type is not None and extracted_media_type != self.media_type:
+            raise ValueError(
+                f"Extracted media type `{extracted_media_type}` does not match "
+                f"expected media type `{self.media_type}`."
+            )
```

Comment:
I suppose these docs should be updated to reflect the new exception case:

https://github.com/WordPress/openverse/blob/0ea4c6cbfa6037c5883207279e891ecbd3e00c83/catalog/dags/common/storage/media.py#L115-L116

Reply:
The `filetype` is still not required generally. I'm just ensuring it raises when a type conflict is found so we know which DAG is ingesting those and adjust it, like the case with Wikimedia. If

Reply:
I mean more that "raises an error" part. The function raises in an error in more than just the conditions related to the required fields now.